### PR TITLE
JIRA:VZ-1771 refactoring associated with using one oci config secret

### PR DESCRIPTION
**Pull request by jmaron99**
In addition to the code refactorings there is a script for generating the OCI configuration secret.

### CODE REVIEW

File: controllers/verrazzano_controller.go [hunk 1]
```diff
@@ -240,6 +251,24 @@ func (r *VerrazzanoReconciler) createInstallJob(ctx context.Context, log logr.Lo
 	return err
 }
 
+func getDNSAuth(r *VerrazzanoReconciler, dns installv1alpha1.DNS) (*installjob.DNSAuth, error) {
+	if dns.OCI != (installv1alpha1.OCI{}) {
+		secret := &corev1.Secret{}
+		err := r.Get(context.TODO(), types.NamespacedName{Name: dns.OCI.OCIConfigSecret, Namespace: "default"}, secret)
```
Comment (bobdonat) [inline controllers/verrazzano_controller.go:257]: I think the namespace should be the same namespace as the verrazzano resource and not hard coded to "default".

**Comment by mgianatagh** [comment]
Should we add some temporary instructions in PLATFORM-OPERATOR-README to describe how to create the secret for OCI DNS?

File: controllers/verrazzano_controller.go (hunk 1 above)
Comment (jmaron99) [reply controllers/verrazzano_controller.go]: fixed

**Comment by jmaron99** [comment]
> Should we add some temporary instructions in PLATFORM-OPERATOR-README to describe how to create the secret for OCI DNS?

took a stab at adding some steps

**Review by mgianatagh — APPROVED** [review]

**Review by bobdonat — APPROVED** [review]
Looks good to me.

**Pull request merged**